lms/djangoapps/courseware/block_render.py

```diff
@@ -587,47 +587,49 @@ def inner_get_block(descriptor):
-            'rebind_user': rebind_user_service,
-            'completion': CompletionService(user=user, context_key=course_id)
-            if user and user.is_authenticated
-            else None,
-            'i18n': XBlockI18nService,
-            'library_tools': LibraryToolsService(store, user_id=user.id if user else None),
-            'partitions': PartitionService(course_id=course_id, cache=DEFAULT_REQUEST_CACHE.data),
-            'settings': SettingsService(),
-            'user_tags': UserTagsService(user=user, course_id=course_id),
-            'badging': BadgingService(course_id=course_id, modulestore=store) if badges_enabled() else None,
-            'teams': TeamsService(),
-            'teams_configuration': TeamsConfigurationService(),
-            'call_to_action': CallToActionService(),
-            'publish': EventPublishingService(user, course_id, track_function),
-        },
-        descriptor_runtime=descriptor._runtime,  # pylint: disable=protected-access
-        request_token=request_token,
-    )
+    services = {
+        'fs': FSService(),
+        'field-data': field_data,
+        'mako': mako_service,
+        'user': user_service,
+        'verification': XBlockVerificationService(),
+        'proctoring': ProctoringService(),
+        'milestones': milestones_helpers.get_service(),
+        'credit': CreditService(),
+        'bookmarks': BookmarksService(user=user),
+        'gating': GatingService(),
+        'grade_utils': GradesUtilService(course_id=course_id),
+        'user_state': UserStateService(),
+        'content_type_gating': ContentTypeGatingService(),
+        'cache': CacheService(cache),
+        'sandbox': SandboxService(contentstore=contentstore, course_id=course_id),
+        'xqueue': xqueue_service,
+        'replace_urls': replace_url_service,
+        'rebind_user': rebind_user_service,
+        'completion': CompletionService(user=user, context_key=course_id)
+        if user and user.is_authenticated
+        else None,
+        'i18n': XBlockI18nService,
+        'library_tools': LibraryToolsService(store, user_id=user.id if user else None),
+        'partitions': PartitionService(course_id=course_id, cache=DEFAULT_REQUEST_CACHE.data),
+        'settings': SettingsService(),
+        'user_tags': UserTagsService(user=user, course_id=course_id),
+        'badging': BadgingService(course_id=course_id, modulestore=store) if badges_enabled() else None,
+        'teams': TeamsService(),
+        'teams_configuration': TeamsConfigurationService(),
+        'call_to_action': CallToActionService(),
+        'publish': EventPublishingService(user, course_id, track_function),
+    }
+
+    descriptor.runtime.get_block_for_descriptor = inner_get_block
+
+    # TODO: When we merge the descriptor and module systems, we can stop reaching into the mixologist (cpennington)
+    descriptor.runtime.mixins = descriptor.runtime.mixologist._mixins  # lint-amnesty, pylint: disable=protected-access
```

Comment:
No, it isn't, since the runtime [already has those mixins](https://github.com/openedx/XBlock/blob/9cf2dfb1e33b2c73dc053b49e7a5cba3293db83b/xblock/runtime.py#L568)